src/dicom-checksum.cpp

```diff
@@ -0,0 +1,29 @@
+#include <core.h>
+#include <db-interface.h>
+#include <string>
+#include <dicom-checksum.h>
+
+std::unordered_map<std::string, std::tuple<std::string, int>> DicomChecksum::checksum_map;
+
+void DicomChecksum::update_checksum(const char *instanceId) {
+    // First we need to retrieve the internalid of the instance using instanceid from 'resource'
+    // Then, using the internalid, retrieve file uuid from 'attachedfiles'
+    std::string uuid = DBInterface::get_uuid_from_instanceid(instanceId);
+    char msg[1024] = {0};
+    sprintf(msg, "instanceId: %s, UUID: %s", instanceId, uuid.c_str());
+    DEBUG_LOG(1, msg);
+
+    // Retrieve md5 from map
+    std::tuple<std::string, int> checksum = DicomChecksum::checksum_map.at(uuid);
+    DicomChecksum::checksum_map.erase(uuid);
```

Comment:
You should use `find(uuid)` which will return an iterator so that `erase` and `at` don't both need to find the entry. a la
```cpp
auto iter = ...find(uuid);
// structured bindings that I'm not sure will work.. but probably will
auto &[key,value] = iter;
auto &[md5,size] = value;
// use md5/size/etc.
...erase(iter);
```